static/app/views/alerts/rules/issue/index.tsx

```diff
@@ -307,41 +345,75 @@ class IssueRuleEditor extends AsyncView<Props, State> {
     }
   };
 
-  fetchPreview = async () => {
+  fetchPreview = (resetCursor = false) => {
     const {organization} = this.props;
-    const {project, rule} = this.state;
+    const {project, rule, previewCursor} = this.state;
 
-    if (!rule) {
+    if (!rule || !organization.features.includes('issue-alert-preview')) {
       return;
     }
+
     this.setState({loadingPreview: true});
-    try {
-      const response = await this.api.requestPromise(
-        `/projects/${organization.slug}/${project.slug}/rules/preview`,
-        {
-          method: 'POST',
-          data: {
-            conditions: rule?.conditions || [],
-            filters: rule?.filters || [],
-            actionMatch: rule?.actionMatch || 'all',
-            filterMatch: rule?.filterMatch || 'all',
-            frequency: rule?.frequency || 60,
-          },
+    if (resetCursor) {
+      this.setState({previewPage: 0});
+    }
+    // we currently don't have a way to parse objects from query params, so this method is POST for now
+    this.api
+      .requestPromise(`/projects/${organization.slug}/${project.slug}/rules/preview`, {
+        method: 'POST',
+        includeAllArgs: true,
+        query: {
+          cursor: resetCursor ? null : previewCursor,
+          per_page: 5,
+        },
+        data: {
+          conditions: rule?.conditions || [],
+          filters: rule?.filters || [],
+          actionMatch: rule?.actionMatch || 'all',
+          filterMatch: rule?.filterMatch || 'all',
+          frequency: rule?.frequency || 60,
+        },
+      })
+      .then(([data, _, resp]) => {
+        GroupStore.add(data?.data);
+
+        const pageLinks = resp?.getResponseHeader('Link');
+        const issueCount = data?.totalCount;
+        this.setState({
+          previewGroups: data?.data.map(g => g.id),
+          previewError: null,
+          pageLinks: pageLinks ?? '',
+          issueCount: issueCount ?? 0,
+          loadingPreview: false,
+        });
+      })
+      .catch(_ => {
+        this.setState({
+          previewError:
+            'Previews are unavailable for this combination of conditions and filters',
+          loadingPreview: false,
+        });
+      });
+  };
+
+  onPreviewCursor: CursorHandler = (cursor, _1, _2, direction) => {
+    this.setState({
+      previewCursor: cursor,
+      previewPage: this.state.previewPage + direction,
+    });
+  };
+
+  fetchMemberList() {
+    const {project} = this.state;
+
+    if (this.props.organization.features.includes('issue-alert-preview')) {
+      fetchOrgMembers(this.api, this.props.organization.slug, [project.id]).then(
```

Comment:
are we not already fetching org members on this page?

Reply:
https://github.com/getsentry/sentry/blob/master/static/app/views/alerts/builder/projectProvider.tsx#L50 they should already be loaded

Reply:
looks like `MemberListStore` stores `User`s, which don't have a project field

Reply:
yeah that looks weird, might be able to get it back from the MemberListStore anyway if you cast the type to `Member` would be nice to only make this request once somehow. If we need to pass a memberList down from the projectProvider that might also be an option

Reply:
lmk if [this](https://github.com/getsentry/sentry/pull/40338/commits/582623577868ff4442e8db52606780938895a30c) works